ADBench/plot_graphs.py

```diff
@@ -446,19 +483,88 @@ def generate_graph(figure_idx, graph_function_type):
     if not do_show:
         pyplot.close(figure)
 
+def get_plot_data(figure_info, sorted_vals_by_tool):
+    '''Adds information to the plot data.'''
+
+    return {
+        "build": figure_info.build_type,
+        "objective": figure_info.objective,
+        "function_type": figure_info.function_type,
+        "test_size": figure_info.maybe_test_size,
+        "values": [
+            {
+                "tool": tool_name,
+                "time": time_vals,
+                "variable_count": var_count_vals,
+                "violations": violation_vals
+            }
+            for tool_name, var_count_vals, time_vals, violation_vals in sorted_vals_by_tool
+        ]
+    }
+
+def extract_vals_and_figure_info_from_plot_data(data):
+    '''Extracts vals sorted by the tool and the figure info from the plot
+    data.'''
+
+    sorted_vals_by_tool = [
+        (val["tool"], val["variable_count"], val["time"], val["violations"])
+        for val in data["values"]
+    ]
+
+    figure_info = namedtuple(
+        "figure_info",
+        "idx, build_type, objective, maybe_test_size, function_type"
+    )
+
+    figure_info.build_type = data["build"]
+    figure_info.objective = data["objective"]
+    figure_info.maybe_test_size = data["test_size"]
+    figure_info.function_type = data["function_type"]
+
+    return sorted_vals_by_tool, figure_info
+
 def main():
     print_messages()
 
+    plot_data = []      # holds all the data that is plotted in the graph
```

Comment:
At the moment the logic for generating plot data is coupled to graph generation. They should be separated.  The logic for this section should look more like

1. `if use_file` then get `plot_data` from file, otherwise generate `plot_data` by looping over all graphs
2. Unconditionally generate the graphs from `plot_data`